tests/typeahead-filter.js

```diff
@@ -29,18 +29,130 @@ describe('typeaheadFilter', function() {
     );
 
-    this.typeaheadFilter = new TypeaheadFilter('.js-typeahead-filter', typeahead.datasets.committees);
+    this.typeaheadFilter = new TypeaheadFilter('.js-typeahead-filter', typeahead.datasets.committees, true);
   });
 
   it('should initialize', function() {
     var typeahead = this.typeaheadFilter.$body.find('.twitter-typeahead');
     expect(typeahead.length).to.equal(1);
   });
 
+  it('should set firstItem', function() {
+    this.typeaheadFilter.setFirstItem({}, {id: 'smith'});
+    expect(this.typeaheadFilter.firstItem.id).to.equal('smith');
+  });
+
+  it('should append checkbox and clear datum on typeahead:select', function() {
+    var appendCheckbox = sinon.spy(this.typeaheadFilter, 'appendCheckbox');
+    var datum = {
+      name: 'FAKE CANDIDATE',
+      id: '12345',
+      office: 'Senate'
+    };
+    this.typeaheadFilter.handleSelect({}, datum);
+
+    expect(appendCheckbox).to.have.been.calledWith({
+      name: 'committee_id',
+      label: 'FAKE CANDIDATE (12345)',
+      value: '12345',
+      id: 'committee_id-12345-checkbox'
+    });
+    expect(this.typeaheadFilter.datum).to.equal(null);
+
+    this.typeaheadFilter.appendCheckbox.restore();
+  });
+
+  it('should set this.datum on typeahead:autocomplte', function() {
+    this.typeaheadFilter.handleAutocomplete({}, {id: '12345'});
+    expect(this.typeaheadFilter.datum).to.deep.equal({id: '12345'});
+  });
+
+  it('should submit on enter', function() {
+    var handleSubmit = sinon.spy(this.typeaheadFilter, 'handleSubmit');
+    this.typeaheadFilter.handleKeypress({keyCode: 13});
+    expect(handleSubmit).to.have.been.calledWith({keyCode: 13});
+    this.typeaheadFilter.handleSubmit.restore();
+  });
+
+  it('should clear the input on blur if it does not allow free text', function() {
+    this.typeaheadFilter.datum = null;
+    this.typeaheadFilter.allowText = false;
+    this.typeaheadFilter.$field.focus();
+    this.typeaheadFilter.$field.typeahead('val','text');
+    this.typeaheadFilter.$field.blur();
+    expect(this.typeaheadFilter.$field.typeahead('val')).to.equal('');
+  });
+
+  it('should enable and disable button when the input changes', function() {
+    var enableButton = sinon.spy(this.typeaheadFilter, 'enableButton');
+    var disableButton = sinon.spy(this.typeaheadFilter, 'disableButton');
+
+    this.typeaheadFilter.$field.typeahead('val', 'FAKE CANDIDATE').change();
+    expect(enableButton).to.have.been.called;
+
+    this.typeaheadFilter.$field.typeahead('val', '').change();
+    expect(disableButton).to.have.been.called;
+
+    this.typeaheadFilter.enableButton.restore();
+    this.typeaheadFilter.disableButton.restore();
+  });
+
+  it('should clear input', function() {
+    this.typeaheadFilter.$field.val('hello');
+    this.typeaheadFilter.enableButton();
+    this.typeaheadFilter.clearInput();
+    expect(this.typeaheadFilter.$field.val()).to.equal('');
+    expect(this.typeaheadFilter.$button.hasClass('is-disabled')).to.be.true;
```

Comment:
To get a more descriptive error, you could write this with a string `match`. e.g.

```javascript
expect(this.typeaheadFilter.$button.class()).to.match(/\bis-disabled\b/);
```

Which would give the error message,
> expected "button button-thing" to match "\bis-disabled\b" but it did not.

If you're not familiar, `\b` matches a word boundary.